docs/architecture/design-proposals/dashboard/deployment-improvements.md

```diff
@@ -0,0 +1,197 @@
+          "properties": {
+            "memory": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["128Mi", "512Mi", "1Gi"] }
+              ]
+            },
+            "cpu": {
+              "type": "string",
+              "anyOf": [
+                { "pattern": ".*" },
+                { "enum": ["100m", "300m", "500m"] }
+              ]
+            }
+          }
+        },
+      }
+    },
+    "enableMetrics": { "type": "boolean", "title": "Enable Metrics" },
+    "replicas": { "type": "integer", "title": "Number of Replicas"}
+  },
+
+  "required": ["wordpressUsername", "wordpressEmail", "persistence", "resources"],
+
+  "kubeapps": {
+    "username": "wordpressUsername",
+    "password": "wordpressPassword",
+    "email": "wordpressEmail",
+    "enableExternalDatabase": "mariadb.enabled",
+    "externalDatabase": "externalDatabase",
+    "disk": "persistence.size",
+    "resources": "resources",
+    "enableMetrics": "enableMetrics",
+    "replicas": "replicas"
+  }
+}
+```
+
+This `values.schema.json` is a valid JSON schema for the `values.yaml` of WordPress. It can be more complete since it's possible to include every single parameter of the values.yaml file.
+
+In order to be able to provide a richer presentation of the form we are including the key `kubeapps`. This allows us to know which parameters we need to represent in the form and how to map them to valid values. In case there is a parameter that is not known we should be able to parse it as a generic input.
+
+## Deployment Workflow
+
+This section defines how the deployment logic will work based on the JSON Schema definition above.
+
+1. The deployment component loads the details of the Chart. If the chart doesn't contain the file `values.schema.json`, continue as today; Only the advanced deployment is allowed. In other case, go to the next step.
+2. Load the `values.schema.json`. If it doesn't contain the `kubeapps` key, auto-generate the form using the tool `react-jsonschema-form`. From this point, two different tabs will be shown to the user. If the user changes tab from `basic` to `advanced` the new form will be shown but the values won't be ported. If the `kubeapps` key is defined go to the next step.
+3. Kubeapps will support ceratin types of parameters (like `username` or `resources`). If those parameters are present, we will render the rich view of the parameter, for example card-choice instead of a text-box. For parameters not supported, we should be able to fallback to the automatic generation. This may not be possible so in that case the parameter will be ignored (with an error for the developer console). If a parameter has a `title` or a `description`, those should be used in the form.
+4. Once the user clicks on "submit", we can use the `kubeapps` mapping to generate a JSON object. We can also use the JSON schema to validate the the processed object complies with the definition.
```

Comment:
This could be a first step - to simply support validating the values.yaml based on the json schema.